src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ConstructorParametersShouldMatchPropertyAndFieldNames.cs

```diff
@@ -0,0 +1,292 @@
+                                                                             isPortedFxCopRule: false,
+                                                                             isDataflowRule: false);
+        internal static DiagnosticDescriptor FieldPublicRule = DiagnosticDescriptorHelper.Create(RuleId,
+                                                                             s_localizableTitle,
+                                                                             s_localizableMessageFieldPublic,
+                                                                             DiagnosticCategory.Design,
+                                                                             RuleLevel.BuildWarning,
+                                                                             description: s_localizableDescriptionFieldPublic,
+                                                                             isPortedFxCopRule: false,
+                                                                             isDataflowRule: false);
+        internal static DiagnosticDescriptor UnreferencedParameterRule = DiagnosticDescriptorHelper.Create(RuleId,
+                                                                             s_localizableTitle,
+                                                                             s_localizableMessageUnreferencedParameter,
+                                                                             DiagnosticCategory.Design,
+                                                                             RuleLevel.BuildWarning,
+                                                                             description: s_localizableDescriptionUnreferencedParameter,
+                                                                             isPortedFxCopRule: false,
+                                                                             isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(PropertyNameRule, FieldRule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterCompilationStartAction((compilationStartContext) =>
+            {
+                INamedTypeSymbol? jsonConstructorAttributeNamedSymbol = compilationStartContext.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemTextJsonSerializationJsonConstructorAttribute);
+                if (jsonConstructorAttributeNamedSymbol == null)
+                {
+                    return;
+                }
+
+                var paramAnalyzer = new ParameterAnalyzer(jsonConstructorAttributeNamedSymbol);
+
+                compilationStartContext.RegisterSymbolStartAction((symbolStartContext) =>
```

Comment:
NIT: the repo coding guideline
```suggestion
                compilationStartContext.RegisterSymbolStartAction((context) =>
```

Reply:
Thanks for the explanation. Done.